### DISCUSSION TITLE
How does ThemeProvider work?

### DISCUSSION THREAD

**Discussion by mariovillamizar** [react-native-elements/react-native-elements#3630] [Q&A]
I'm trying to use ThemeProvider to switch between dark/light appearance. So I'm changing the mode from Settings (iOS) and I'm expecting for the app to change the theme, but it's working really weird:

This is my environment:
```json
"react": "18.0.0",
"react-native": "0.69.5",
"@rneui/base": "^4.0.0-rc.6",
"@rneui/themed": "^4.0.0-rc.6",
"typescript": "^4.8.2",
"node": "16.17.0"
```

This is my code:

```typescript
import React, { useEffect } from 'react';
import { useColorScheme } from 'react-native';
import { Button, createTheme, ThemeProvider } from '@rneui/themed';

const lightTheme = createTheme({
  components: {
    Button: {
      color: 'blue'
    }
  }
});

const darkTheme = createTheme({
  components: {
    Button: {
      color: 'yellow'
    }
  }
});

function App() {
  useEffect(() => {
    console.log('Current environment is:', Config.ENV);
  }, []);

  const colorScheme = useColorScheme();
  useEffect(() => {
    console.log(colorScheme);
  }, [colorScheme]);

  return (
    <ThemeProvider theme={colorScheme === 'light' ? lightTheme : darkTheme}>
      <Button title='My Button' />
    </ThemeProvider>
  );
}

export default App;
```

As you can see, I have a log to check if the colorScheme is changing, and I can confirm it's doing its work, so that's no the problem.
The initial state in 'light' mode, it looks like this:
![Simulator Screen Shot - iPhone 13 - 2022-09-06 at 21 41 13](https://user-images.githubusercontent.com/10704258/188776619-fd549e4a-7974-4b89-aaec-c5a84c415d60.png)

And when I go to Settings, change the appearance to dark, and go back to the app, it looks like this:
![Simulator Screen Shot - iPhone 13 - 2022-09-06 at 21 42 40](https://user-images.githubusercontent.com/10704258/188776793-c8d208d6-bb47-497b-a96d-b27837866dcb.png)

As you can see the background of the view is changing but the color the button keeps blue.

So, what am I doing wrong?
Thanks in advance.

**Comment by arpitBhalla** [comment]
You don't need to use separate theme for light and dark mode, refer the [themeProvider docs](https://reactnativeelements.com/docs/next/customization/themeprovider)
